### PR TITLE
[plugin.video.intergalacticfm] 2.0.1

### PR DESCRIPTION
**Pull request by dromer**
### Description
We've completely rewritten our video plugin so that we can load available channels from a resource on our website. We also include all meta-data such as images and descriptions via the plugin.

We will regularly update the plugin with new channels and updated data (but not too often!).
Eventually we want to include a way to update this data internally so we only have to do a PR when there are major changes.

### Checklist:
- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

**Comment by dromer** [comment]
I know it now took 2 commits for this PR. Should I close it and do a fresh one, or is it ok?

cheers

File: plugin.video.intergalacticfm/README.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+3. Choose **Install from repository**
+4. Choose **Video add-ons**
+5. Navigate to **Intergalactic FM TV** and select it
+6. Choose **Install**
+
+### Linux
+
+Install Kodi with
+
+    sudo apt-get install kodi kodi-repository-kodi
```
Comment (Rechi) [inline plugin.video.intergalacticfm/README.md]: `kodi-repository-kodi` is only required if you want to use the broken Debian package. Remove it or mention it only in a note.

File: plugin.video.intergalacticfm/addon.xml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="plugin.video.intergalacticfm" name="Intergalactic FM TV" version="2.0.1" provider-name="Dreamer and Pander">
```
Comment (Rechi) [inline plugin.video.intergalacticfm/addon.xml]: The correct separator for multiple devs is a comma (`,`).
```suggestion
<addon id="plugin.video.intergalacticfm" name="Intergalactic FM TV" version="2.0.1" provider-name="Dreamer, Pander">
```

File: plugin.video.intergalacticfm/addon.xml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        <fanart>resources/fanart.jpg</fanart>
+        <screenshot>screenshots/s1.png</screenshot>
+        <screenshot>screenshots/s2.png</screenshot>
+        <screenshot>screenshots/s3.png</screenshot>    
+    </assets>
+    <news>Updated with latest artwork metadata</news>
+    <license>The GNU General Public License (GPL)</license>
+    <source>https://github.com/intergalacticfm/plugin.video.intergalacticfm</source>
+    <website>http://intergalactic.fm</website>
+    <forum>http://forum.xbmc.org/showthread.php?tid=184904</forum>
```
Comment (Rechi) [inline plugin.video.intergalacticfm/addon.xml]:
```suggestion
    <forum>https://forum.kodi.tv/showthread.php?tid=257201</forum>
```

File: plugin.video.intergalacticfm/main.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+_addon = xbmcaddon.Addon()
+
+
+fm = 'https://www.intergalactic.fm/'
+tv = 'https://www.intergalactic.tv/'
+pn = 'ifm-system/playingnow.json'
+pl = 'playlist.m3u8'
+
+
+#TODO If JSON on GitHub is younger, load that file. Otherwise, load local file.
```
Comment (Rechi) [inline plugin.video.intergalacticfm/main.py]: This won't be allowed in our repo.

Comment (dromer) [reply plugin.video.intergalacticfm/main.py]: Ok, so the only way to push updates of this kind of meta-data would be to update the plugin? We could also host this json somewhere else and pull from there. Why is it not allowed to do this from github?

Comment (Rechi) [reply plugin.video.intergalacticfm/main.py]: The reason is that we want to review those changes. The only place allowed for online requests are domains where the content is located. In your case intergalactic.fm and intergalactic.tv.

**Comment by dromer** [comment]
Hmm, tried to squash the commits, but still end up with 2 of them ..

File: plugin.video.intergalacticfm/main.py (hunk 4 above)
Comment (dromer) [reply plugin.video.intergalacticfm/main.py]: Understood. If we ever implement this it'll be through our domains.

**Comment by Rechi** [comment]
To fix your pull request execute the following commands (origin is your repo, upstream is the xbmc repo)

```
git fetch --all
git checkout krypton
git reset --hard upstream/krypton
git checkout origin/krypton -- plugin.video.intergalacticfm/
git commit -m '[plugin.video.intergalacticfm] 2.0.1'
git push -f origin krypton
```

**Comment by dromer** [comment]
Thnx @Rechi , but somehow my repo is not referencing to upstream.

In our forked repo I get:

```
repo-plugins$ git reset --hard upstream/krypton
fatal: ambiguous argument 'upstream/krypton': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

The .git/config is:
```
repo-plugins$ cat .git/config 
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
[remote "origin"]
	url = git@github.com:intergalacticfm/repo-plugins.git
	fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
	remote = origin
	merge = refs/heads/master
[remote "plugin.video.intergalacticfm"]
	url = git@github.com:intergalacticfm/plugin.video.intergalacticfm.git
	fetch = +refs/heads/*:refs/remotes/plugin.video.intergalacticfm/*
[branch "plugin.video.intergalacticfm"]
	remote = plugin.video.intergalacticfm
	merge = refs/heads/master
[branch "krypton"]
	remote = origin
	merge = refs/heads/krypton
```

**Comment by Rechi** [comment]
`git remote add upstream git@github.com:xbmc/repo-plugins.git`

**Comment by dromer** [comment]
Heuhm, same message.

**Comment by dromer** [comment]
There we go. Thnx @Rechi !
(one day I'll get the hang of this git thing!)

**Comment by Rechi** [comment]
@dromer is the version at https://github.com/xbmc/repo-plugins/tree/isengard/plugin.video.intergalacticfm still working?
If not please submit a version update to that branch where you mark the add-on broken.

**Comment by dromer** [comment]
@Rechi yes the old plugin still works. just that we rewrote it and tested only on krypton

old versions for older kodi's should still be fine at the moment

**Pull request merged**